### ISSUE TITLE
Custom quests give too low gold reward at 60

### ISSUE THREAD

**Issue by sieleyah** [slowtorta/turtlewow-bug-tracker#2439]
**Please, provide a clear description what the bug is:**
Hi, this seems to be a general issue with turtle-wow custom quest lines, while other lvl 40+ quests award 1-2gold upon completing at max level and 50+ are usually between 2-3 gold, the quests added by server in these higher level ranges mostly award just 30-40silver instead. (Note: This is not merely the quest's own monetary reward, but also the compensation you get for not earning exp anymore. I suspect the latter portion is scaling wrong.)
While these are custom quests, I'm certain the developers intended them to be just as rewarding to complete as any other quest so this must be a bug.

**Steps to reproduce the behavior with as much detail as possible:**
Complete any high level custom quests as a lvl 60, such as those at Karfang Hold, or the Sandfury troll ones in Tanaris. Then compare the gold reward to other quests in the same level bracket, and same quality (i.eg. dungeon, elite, etc)
They will all give a fraction of the gold that original vanilla quests award.

**IDs of affected creatures, items, quests or spells with a link to the relevant page:**

It's really too many quests to link. It is a general issue.

**Expected behavior. Describe how it should work:**
I'd wager quests added by turtle-wow should have largely similar money rewards when completed at lvl 60 as those originally in the game.

**Comment by Sringaram** [comment]
The same applies to basically all custom loot and custom loot from custom dungeon.

**Comment by slowtorta** [comment]
I'm sorry, but this kind of report isn't something we can work with. A general overview of the custom quests in the DB does not indicate any major differences. If you spot a quest that you think gives too less gold compared to a quest of the same difficulty, please open a  report about this specific quest or a set of quests!

**Comment by cyaohiri** [comment]
The formula for getting gold at max level is: `Money in copper = XP * 6`. You can use this to calculate if any of the quest money rewards are too low at level 60. Source: https://vanilla-wow-archive.fandom.com/wiki/Experience_point#Money_Instead_of_XP_at_Max_Level

Could you give examples of some quests you've done and how much gold you've gained?

**Comment by slowtorta** [comment]
Some custom quests are just go-talk-to kinds of quests, they aren't supposed to give money rewards. Applying some generalized fix will cause us more problems.

**Comment by sieleyah** [comment]
Ok here is the example and why this is a general issue:

Quest "Forest troll scum" in Karfang Hold sends you to kill War Master Voone. This is a lvl 60 dungeon boss quest that currently awards about 36 silver to a lvl 60 character instead of the 3-4 gold you would expect.

After checking the db and the formula linked by cyaohiri the reason is clear: In the DB this quest has 2 different exp values shown. The first one on the right side "RevXP" is 6400 which is the intended exp reward for the quest. The second one, in the quest description says this quest awards 640 exp and 250 rep upon completion. This number is 1/10 of the actual intended value and I suspect the gold at 60 is calculated from this value as it roughly checks out with the x6 formula, giving 36 silver instead of 3,6 gold.

You can't expect me to go through roughly 200 custom quest and report each 1 by 1 when this is a general issue that seems to stem from a mismatch of values in DB.

**Comment by cyaohiri** [comment]
> In the DB this quest has 2 different exp values shown. The first one on the right side "RevXP" is 6400 which is the intended exp reward for the quest. The second one, in the quest description says this quest awards 640 exp and 250 rep upon completion. This number is 1/10 of the actual intended value.

That seems to be a common thing in the database. This applies to both custom and vanilla quests. Sometimes the value isn't exactly one tenth of the actual value, probably due to rounding. For example [The Missing Diplomat](https://database.turtle-wow.org/?quest=1244) gives 2450 XP, but under the quest description it says 250.

**Comment by jameyboor** [comment]
![afbeelding](https://user-images.githubusercontent.com/1365563/202000730-cb6b161a-22c7-43c0-b97a-a19070c05a23.png)

The Missing Diplomat gives 2450 XP as stated in the DB. The left-hand side on the DB website showing XP is incorrect and will be fixed shortly. 

Furthermore, gold on max level is calculated from one data point only, RewMoneyMaxLevel. It's also in the DB on the right. If you compare the values of our level 60 custom quests to Blizz quests it looks like ours actually give out more than the normal level 60 Blizz quests. The only quests that are NOT up to par are dungeon quests that are custom. We will look into this.

**Comment by jameyboor** [comment]
DB now shows correct XP values.

**Comment by sieleyah** [comment]
Thank you. As I said this report is mostly for the high level quest hubs (40+) and particularly when questing at lvl 60 for gold and these hubs indeed have many dungeon quests too so it checks out with your assessment.

I don't recall this issue when questing as a lowbie in low level areas. (some level 10-20 turtle quests indeed give more exp and silver than others in the same bracket)

**Comment by sieleyah** [comment]
Is this fix live? I just tested after restart and the custom quests in blasted still give 30silver each. 
They show reward on database now, but it's still calculated wrong. 10% of the intended value.

**Comment by cyaohiri** [comment]
First of all, since you still haven't given the name of a quest you've done, I'll use [The Garrison Armory Disaster](https://database.turtle-wow.org/?quest=40428) as an example. It's a level 48 quest that gives 4,850 experience. Converted to gold, that would be 4,850 * 6 = 29100 copper or 2.91 gold, whereas the calculated money reward at level 60 is 29.1 silver; 10% the amount.
I suspect the reason for this is that the experience penalty for being a higher level doesn't take into account that here, you receive full experience until more than 25 levels above the quest level, thus incorrectly applying the penalty of around 90% to the experience reward as is done in vanilla, such that a quest level 50, i.e. [The Basilisk's Bite](https://www.wowhead.com/classic/quest=2601/the-basilisks-bite) would yield only 470 experience instead of 4,700, and thus reward 28.2 silver instead of 2.82 gold (see the level slider on Wowhead).

As as example of a vanilla quest that doesn't have the penalty as it's a level 58 quest, see [Target: Gahrron's Withering](https://www.wowhead.com/classic/quest=5225/target-gahrrons-withering). It rewards 6,200 experience, and at 60, it rewards 3.72 gold which is exactly 6200 * 6 in copper.

**Comment by sieleyah** [comment]
It has nothing to do with applying a penalty because it affects same level quests too. Simply the math is off by a digit.
I did give an example above, I named one of the Kargath quests which is a lvl 60 dungeon quest you can take at lvl 48. 
 (i have also just done the Depths of Karazhan II, which is lvl 60 quest you can take at lvl 60, this was done after update, and awards about 37silver)
 I don't understand how the DB can say something awards 6250 xp, and then calculate it awards 37 silver. 6x 6.2k is not 3700 RewMoneyMaxLvl as the DB says to calculate from. Where did one zero go? This is like basic math failed by some algorithm.
 https://database.turtle-wow.org/?quest=40305
 
 And yes this affects nearly every custom quest above lvl 35.

**Comment by sieleyah** [comment]
UPDATE: This issue seems to affect non-custom quests too, particularly lvl 60 raid quests.
I wasn't aware that these quests are supposed to give decent gold, but after some research I found that majority of the lvl 60 raid quests I completed so far award only 10% of the intended gold reward.
Example: https://www.wowhead.com/classic/quest=8710/kris-of-unspoken-names This quest awards 3.9gold on classic, and only 39 silver here. 
Example2: https://www.wowhead.com/classic/quest=8519/a-pawn-on-the-eternal-board the entire quest chain awards only 10% of the intended money reward here

If I add up how much gold I had already been ripped off, it's easily 200~gold at this point...
